default/python/AI/ShipDesignAI.py

```diff
@@ -854,6 +861,15 @@ def update_stats(self, ignore_species=False):
         self.production_cost = local_cost_cache.get(self.hull.name, self.hull.productionCost(fo.empireID(), self.pid))
         self.production_time = local_time_cache.get(self.hull.name, self.hull.productionTime(fo.empireID(), self.pid))
 
+        # calculate hull stat effects
+        if self.species and not ignore_species:
+            self.design_stats.fuel += _get_species_fuel_bonus(self.species)
+        # TODO tech hull fuel effects
+        if self.design_stats.fuel < 0:
+            self.design_stats.fuel = 0
```

Comment:
Is max fuel meter enforced to be >= 0 at all times by the backend?
If not, this will easily lead to wrong calculations if there are negative interim results and fuel parts are used in the design.

Reply:
Backend is not involved here. This only happens for NO_FUEL effect which is the effect happening directly before the fuel efficiency. So if you have NO_FUEL, your fuel will be 0.
I add a source comment.